features/vaadin-surveillance-views/src/test/java/org/opennms/features/vaadin/surveillanceviews/service/NMS15448_IT.java

```diff
@@ -0,0 +1,156 @@
+import org.junit.runner.RunWith;
+import org.opennms.core.spring.BeanUtils;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
+import org.opennms.core.test.db.MockDatabase;
+import org.opennms.core.test.db.annotations.JUnitTemporaryDatabase;
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.netmgt.dao.DatabasePopulator;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsCategory;
+import org.opennms.netmgt.model.OnmsDistPoller;
+import org.opennms.netmgt.model.OnmsEvent;
+import org.opennms.netmgt.model.OnmsSeverity;
+import org.opennms.test.JUnitConfigurationEnvironment;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import com.google.common.collect.Sets;
+
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations = {
+        "classpath:/META-INF/opennms/applicationContext-commonConfigs.xml",
+        "classpath:/META-INF/opennms/applicationContext-soa.xml",
+        "classpath:/META-INF/opennms/applicationContext-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-databasePopulator.xml"
+})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase(reuseDatabase = false, tempDbClass = MockDatabase.class)
+@Transactional
+public class NMS15448_IT implements InitializingBean {
+    @Autowired
+    DatabasePopulator databasePopulator;
+
+    @Autowired
+    TransactionTemplate transactionOperations;
+
+    private OnmsDistPoller distPoller;
+
+    @Override
+    public void afterPropertiesSet() {
+        BeanUtils.assertAutowiring(this);
+    }
+
+    @Before
+    public void setUp() {
+        databasePopulator.populateDatabase();
+        distPoller = databasePopulator.getDistPollerDao().whoami();
+    }
+
+    private OnmsAlarm buildAlarm(final OnmsEvent event, boolean acknowledged) {
+        final OnmsAlarm alarm = new OnmsAlarm();
+        alarm.setDistPoller(distPoller);
+        alarm.setUei(event.getEventUei());
+        alarm.setAlarmType(OnmsAlarm.PROBLEM_TYPE);
+        alarm.setNode(databasePopulator.getNode1());
+        alarm.setDescription("This is a test alarm");
+        alarm.setLogMsg("this is a test alarm log message");
+        alarm.setCounter(1);
+        alarm.setIpAddr(InetAddressUtils.getInetAddress("192.168.1.1"));
+        alarm.setSeverity(OnmsSeverity.NORMAL);
+        alarm.setFirstEventTime(event.getEventTime());
+        alarm.setLastEvent(event);
+        alarm.setServiceType(databasePopulator.getServiceTypeDao().findByName("ICMP"));
+        if (acknowledged) {
+            alarm.setAlarmAckUser("foobar");
+            alarm.setAlarmAckTime(new Date());
+        }
+        return alarm;
+    }
+
+    @Test
+    public void testSomething() {
```

Comment:
Perhaps rename to something like `testSurveillanceViewServiceReturnsOnlyUnacknowledgedAlarms` or similar.

Reply:
Oh my god. I forgot to rename the method. I'm sorry.